lib/membrane/core/element/demand_controller/auto_flow_utils.ex

```diff
@@ -7,13 +7,58 @@ defmodule Membrane.Core.Element.DemandController.AutoFlowUtils do
   }
 
   require Membrane.Core.Child.PadModel, as: PadModel
+  require Membrane.Logger
   require Membrane.Pad, as: Pad
 
   defguardp is_input_auto_pad_data(pad_data)
             when is_map(pad_data) and is_map_key(pad_data, :flow_control) and
                    pad_data.flow_control == :auto and is_map_key(pad_data, :direction) and
                    pad_data.direction == :input
 
+  @spec pause_demands(Pad.ref(), State.t()) :: State.t()
+  def pause_demands(pad_ref, state) do
+    :ok = ensure_auto_input_pad!(pad_ref, :pause_auto_demand, state)
+    set_auto_demand_stopped_flag(pad_ref, true, state)
+  end
+
+  @spec resume_demands(Pad.ref(), State.t()) :: State.t()
+  def resume_demands(pad_ref, state) do
+    :ok = ensure_auto_input_pad!(pad_ref, :resume_auto_demand, state)
+    state = set_auto_demand_stopped_flag(pad_ref, false, state)
+    auto_adjust_atomic_demand(pad_ref, state)
+  end
+
+  defp ensure_auto_input_pad!(pad_ref, action, state) do
+    case PadModel.get_data!(state, pad_ref) do
+      %{direction: :input, flow_control: :auto} ->
+        :ok
+
+      %{direction: :output} ->
+        raise Membrane.ElementError,
+              "Action #{inspect(action)} can only be returned for input pads, but #{inspect(pad_ref)} is an output pad"
+
+      %{flow_control: flow_control} ->
+        raise Membrane.ElementError,
+              "Action #{inspect(action)} can only be returned for pads with :auto flow control, but #{inspect(pad_ref)} pad flow control is #{inspect(flow_control)}"
+    end
+  end
+
+  @spec set_auto_demand_stopped_flag(Pad.ref(), boolean(), State.t()) :: State.t()
+  defp set_auto_demand_stopped_flag(pad_ref, new_value, state) do
```

Comment:
[NIT] I think it will be easier to read it that way
```suggestion
  defp set_auto_demand_stopped_flag(pad_ref, stopped?, state) do
```